Added relations to team model

makarman opened the pull request:
Added one_to_many relation to owner of team
Added many_t_many relation to members of team
Made new migration file



## Code reviewers

- [ ] @dmytrorevak
- [ ] @IvanAndrukh
- [ ] @JjuliaP
- [ ] @makarman
- [ ] @noobhunters
- [ ] @Silent777
- [ ] @Staryj812

### Second Level Review

- [ ] @ssOleg
- [ ] @lhalam



## CHECK LIST
- [x]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

redmyt commented on eventually/authentication/models.py:
```diff
@@ -32,11 +32,11 @@ class CustomUser(AbstractBaseUser):
     updated_at = models.DateTimeField(auto_now=True)
```
this file is not your responsibility

makarman replied:
done

redmyt commented on eventually/eventually/urls.py:
```diff
@@ -13,9 +13,8 @@
 
 urlpatterns = [
-    url(r'^admin/', admin.site.urls),
+    url(r'^api/v1/team/', include('team.urls')),
```
have we needed changes in this file? have we changed them?

makarman replied:
done

redmyt commented on eventually/team/models.py:
```diff
@@ -31,37 +32,74 @@ class Team(models.Model):
     created_at = models.DateTimeField(auto_now_add=True, editable=False)
     updated_at = models.DateTimeField(auto_now=True)
+    owner = models.ForeignKey(CustomUser, null=True)
+    members = models.ManyToManyField(CustomUser, related_name='+')
```
does `members` field need the `related name` keyword?

makarman replied:
yes, because I use CustomUser in ForeignKey and in ManyToManyField

redmyt commented on eventually/team/models.py:
```diff
@@ -31,37 +32,74 @@ class Team(models.Model):
         {
         'id': 1,
-        'name': 'name',
+        'name': 'name',feature/team
```
what is `feature/team`?

redmyt commented on eventually/team/models.py:
```diff
@@ -31,37 +32,74 @@ class Team(models.Model):
         'id': 1,
-        'name': 'name',
+        'name': 'name',feature/team
         'description': 'desc',
```
maybe it is a good idea to write each dict's field from a new line using `|` char.

makarman committed:
Added relations to team model

Added one_to_many relation to owner of team
Added many_to_many relation to members of team
Made new migration file

ssOleg commented on eventually/authentication/models.py:
```diff
@@ -83,7 +83,7 @@ def create(first_name, middle_name, last_name, email, password):
+    def to_dict(self):
         """
         :return: manually-defined attributes of the object. Inherited attributes aren't shown.
         :example: {'id': 8, 'first_name': 'fn', 'middle_name': 'mn',
```
`'FIXME` please replace it with `TODO` or remove.

makarman replied:
it is not my app. I have removed  eventually/authentication/models.py

ssOleg commented on eventually/team/models.py:
```diff
@@ -31,38 +31,75 @@ class Team(models.Model):
-        'created_at': 1509539536196,
-        'updated_at': 1509539536196
-        }
+        | {
```
please write a docstrings in accordance with the project style.

makarman replied:
done

makarman replied:
It is our project style.

ssOleg commented on eventually/team/models.py:
```diff
@@ -110,19 +157,28 @@ def create(name=None, description='', image=''):
         """
 
         team = Team()
+
```
why do you need this empty line?

makarman replied:
done

ssOleg commented on eventually/team/views.py:
```diff
@@ -1,8 +1,12 @@
+class TeamView(View):
+    """Team view handles GET, POST, PUT, DELETE requests.
+    """
+    pass
```
when you have docstring you can remove `pass` I think.

makarman replied:
done
